lib/core/include/irods/library_support.h

```diff
@@ -0,0 +1,15 @@
+#ifndef IRODS_LIBRARY_SUPPORT_H
+#define IRODS_LIBRARY_SUPPORT_H
+
+/// \file
+/// \brief iRODS Feature Test Macros
+///
+/// This file defines macros that can be used to test whether one or more features
+/// are supported by the iRODS library.
+///
+/// \since 4.3.1
+
+#define IRODS_HAS_API_ENDPOINT_SWITCH_USER 4003001L
```

Comment:
Should be date-based as discussed during standup

Reply:
Done.